pkg/registry/apis/dashboard/schema_validation.go

```diff
@@ -0,0 +1,98 @@
+package dashboard
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v0alpha1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v1alpha1"
+	"github.com/grafana/grafana/apps/dashboard/pkg/apis/dashboard/v2alpha1"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+)
+
+// ValidateDashboardSpec validates the dashboard spec and throws a detailed error if there are validation errors.
+func (b *DashboardsAPIBuilder) ValidateDashboardSpec(ctx context.Context, obj runtime.Object, a admission.Attributes) error {
+	// This will be removed with the other PR
+	return nil //nolint:govet
+
+	accessor, err := utils.MetaAccessor(obj)
+	if err != nil {
+		return fmt.Errorf("error getting meta accessor: %w", err)
+	}
+	errorOnSchemaMismatches := true
+	switch mode := getFieldValidationMode(a); mode {
+	case metav1.FieldValidationIgnore:
+		errorOnSchemaMismatches = false
+	case metav1.FieldValidationWarn:
+		return apierrors.NewBadRequest("FieldValidationWarn is not supported")
+	case metav1.FieldValidationStrict:
+		break
+	default:
+		return apierrors.NewBadRequest(fmt.Sprintf("Invalid field validation mode: %s", mode))
+	}
+	if errorOnSchemaMismatches {
+		if _, ok := obj.(*v1alpha1.Dashboard); ok {
+			errorOnSchemaMismatches = !b.features.IsEnabled(ctx, featuremgmt.FlagDashboardDisableSchemaValidationV1)
+		} else if _, ok := obj.(*v2alpha1.Dashboard); ok {
+			errorOnSchemaMismatches = !b.features.IsEnabled(ctx, featuremgmt.FlagDashboardDisableSchemaValidationV2)
+		}
+	}
+
+	alwaysLogSchemaValidationErrors := b.features.IsEnabled(ctx, featuremgmt.FlagDashboardSchemaValidationLogging)
+
+	var errors field.ErrorList
+	var schemaVersionError *field.Error
+	if errorOnSchemaMismatches || alwaysLogSchemaValidationErrors {
```

Comment:
Do we need all of these flags here?

I don't have the full context here of course, but I think we already handle permissive validation using `FieldValidationMode == "Ignore"`. Having more conditions on top of that introduces so much extra complexity into the setup – the amount of possible config permutations is IMHO not worth it.

tl;dr – are we sure we want to be _this_ careful?

And if the answer is "yes we need to be super-duper careful", can we just have 1 feature toggle here (enable schema validation = yes / no).